src/main/java/org/llorllale/cactoos/matchers/IsBytes.java

```diff
@@ -0,0 +1,137 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.llorllale.cactoos.matchers;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.stream.IntStream;
+import org.cactoos.Bytes;
+import org.cactoos.bytes.BytesOf;
+import org.cactoos.bytes.UncheckedBytes;
+import org.cactoos.iterable.IterableOfBytes;
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+/**
+ * Matcher for {@link Bytes}.
+ *
+ * @since 1.0.0
+ * @checkstyle ProtectedMethodInFinalClassCheck (200 lines)
+ */
+public final class IsBytes extends TypeSafeDiagnosingMatcher<Bytes> {
+
+    /**
+     * Expected bytes.
+     */
+    private final UncheckedBytes bytes;
+
+    /**
+     * Ctor.
+     * @param bytes The bytes to match against
+     */
+    public IsBytes(final byte... bytes) {
+        this(new BytesOf(bytes));
+    }
+
+    /**
+     * Ctor.
+     * @param chars The chars to match against
+     */
+    public IsBytes(final char... chars) {
+        this(new BytesOf(chars));
+    }
+
+    /**
+     * Ctor.
+     * @param ints The bytes to match against
+     */
+    public IsBytes(final int... ints) {
+        this(
+            IntStream
+                .of(ints)
+                .collect(
+                    () -> ByteBuffer.allocate(ints.length),
+                    (bts, value) -> bts.put((byte) value),
+                    ByteBuffer::put
+                ).array()
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param string The string to match against
+     */
+    public IsBytes(final String string) {
+        this(new BytesOf(string));
+    }
+
+    /**
+     * Ctor.
+     * @param bytes The bytes to match against
+     */
+    public IsBytes(final Bytes bytes) {
+        this.bytes = new UncheckedBytes(bytes);
+    }
+
+    @Override
+    public void describeTo(final Description description) {
+        description.appendValue(new IterableOfBytes(this.bytes));
+    }
+
+    @Override
+    @SuppressWarnings(
+        {"PMD.AvoidCatchingGenericException", "PMD.AvoidCatchingThrowable"}
+    )
+    protected boolean matchesSafely(
+        final Bytes item, final Description description
+    ) {
+        // @checkstyle IllegalCatchCheck (100 lines)
+        boolean res = true;
+        byte[] presented = new byte[0];
+        try {
+            presented = item.asBytes();
+        } catch (final Exception ex) {
+            description
+                .appendText("thrown ")
+                .appendValue(ex);
+            res = false;
```

Comment:
@andreoss this is not the responsibility of the matcher to check for exceptions: as in all the other matchers, if there is an exception we should propagate it and the test will fail because of it, not need to introduce extra behaviour here.

Reply:
@victornoel The reason I did it this way is to have a nicer report when `Bytes` throws while being matched (`orange` test vs `red` test)